socs/agents/lakeshore372/agent.py

```diff
@@ -787,6 +795,248 @@ def servo_to_temperature(self, session, params):
 
         return True, f'Setpoint now set to {params["temperature"]} K'
 
+    @ocs_agent.param('channel', type=int)
+    @ocs_agent.param('state', type=str, choices=['on', 'off'])
+    def engage_channel(self, session, params):
+        """engage_channel(channel=None)
+
+        **Task** - Enables/disables a channel on the LS372
+
+        Parameters:
+            channel (int): Channel number to enable
+            state (str): Desired power state of channel; 'on' or 'off'
+        """
+        with self._lock.acquire_timeout(job='engage_channel') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self._lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            session.set_status('running')
+
+            channel = params['channel']
+            state = params['state']
+            if state == 'on':
+                self.module.channels[channel].enable_channel()
+            else:
+                self.module.channels[channel].disable_channel()
+
+        return True, "Channel {} powered {}".format(channel, state)
+
+    @ocs_agent.param('channel', type=int)
+    @ocs_agent.param('curve_number', type=int)
+    def set_calibration_curve(self, session, params):
+        """set_calibration_curve(channel=None, curve_number=None)
+
+        **Task** - Sets the calibration curve number for a particular channel.
+
+        Parameters:
+            channel (int): Channel number to set calibration curve to
+            curve_number (int): Curve number of calibration curve uploaded to
+                the LS372
```

Comment:
I limited the curve numbers to 21-59 since 1-20 have Lakeshore's standard curve locations and standard curve can't be changed by the user. The manual does say that not all 20 slots are filled but figure 21-59 is the best bet.